### PR TITLE
Update `DifficultyCalculator` to take an `IRulesetInfo` and `IWorkingBeatmap`

### PR DESCRIPTION
**Pull request by peppy**
- [x] Depends on #15631.
- [x] Depends on #15594.

### CODE REVIEW

File: osu.sln.DotSettings [hunk 1]
```diff
@@ -209,6 +209,8 @@
 	<s:String x:Key="/Default/CodeInspection/Highlighting/InspectionSeverities/=StructCanBeMadeReadOnly/@EntryIndexedValue">WARNING</s:String>
 	<s:String x:Key="/Default/CodeInspection/Highlighting/InspectionSeverities/=SuggestVarOrType_005FBuiltInTypes/@EntryIndexedValue">SUGGESTION</s:String>
 	<s:String x:Key="/Default/CodeInspection/Highlighting/InspectionSeverities/=SuggestVarOrType_005FSimpleTypes/@EntryIndexedValue">DO_NOT_SHOW</s:String>
+	<s:String x:Key="/Default/CodeInspection/Highlighting/InspectionSeverities/=SuspiciousTypeConversion_002EGlobal/@EntryIndexedValue"></s:String>
+	<s:Boolean x:Key="/Default/CodeInspection/Highlighting/InspectionSeverities/=SuspiciousTypeConversion_002EGlobal/@EntryIndexRemoved">True</s:Boolean>
```
Comment (smoogipoo) [inline osu.sln.DotSettings:213]: Does this remove the inspection? I thought we decided on having it error/warning instead?

Comment (peppy) [reply osu.sln.DotSettings:213]: That's on the base PR, but it does look to correctly warn at my end:

20211115 200404 (JetBrains Rider-EAP)

on master:
20211115 200404 (JetBrains Rider-EAP)

Comment (smoogipoo) [reply osu.sln.DotSettings:213]: Yep, confirmed that it works. :psyduck:

**Review by bdach — COMMENTED** [review]
source seems sane except for prereq concerns

**Review by smoogipoo — APPROVED** [review]

**Pull request merged**